invenio_rdm_records/records/systemfields/statistics.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 TU Wien.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Cached transient field for record statistics."""
+
+from invenio_records.systemfields import SystemField
+
+from ...stats.utils import get_record_stats
+
+
+class RecordStatistics(SystemField):
+    """Field for lazy fetching and caching (but not storing) of record statistics."""
+
+    def __init__(self, key=None):
+        """Constructor."""
+        super().__init__(key=key)
+
+    #
+    # Data descriptor methods (i.e. attribute access)
+    #
+    def __get__(self, record, owner=None):
+        """Get the persistent identifier."""
+        if record is None:
+            # returns the field itself.
+            return self
+
+        stats = record.get("stats", None)
+        if stats:
+            return stats
+
+        recid = record["id"]
+        parent_recid = record.parent["id"]
+        stats = get_record_stats(recid=recid, parent_recid=parent_recid)
```

Comment:
On the point of the out-of-sync stats on the search vs. record landing page: on Zenodo we actually [pull the stats on the record landing page from ES](https://github.com/zenodo/zenodo/blob/513e1fc8882c80e4568ab66bae60ebc658646356/zenodo/modules/stats/utils.py#L119-L124), buy fetching only the `_stats` key from the indexed document. That way we have the same numbers, and ES acts as a "cache" as well. We could maybe use the same logic here?

Reply:
I've updated the statistics system field s.t. it will first try to get the record's stats from the records search index (`rdmrecords-records-record-v5.0.0`).
Only as a fallback will the system field try to get the most up-to-date statistics via the `invenio-stats` queries.

The code isn't exactly pretty, but I don't think that the required piece of logic has been implemented already (e.g. the `service.indexer` doesn't have a `.get()`, etc.).